File: app2_testing/app/src/main/java/com/example/app2_testing/MainActivity.java
package com.example.app2_testing;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button button = (Button) findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String inp1 = ((EditText) findViewById(R.id.editText1)).getText().toString();
                String inp2 = ((EditText) findViewById(R.id.editText2)).getText().toString();
                ((TextView) findViewById(R.id.textView)).setText(concatString(inp1, inp2));
            }
        });
    }

    public static String concatString(String inp1, String inp2) {
        return inp1 + " - " + inp2;
    }
}

File: app7_cross_platform/app7_native/app/src/main/java/id/ac/ui/cs/mobileprogramming/kevinlh/app7_native/MainActivity.kt
package id.ac.ui.cs.mobileprogramming.kevinlh.app7_native

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.floatingactionbutton.FloatingActionButton

class MainActivity : AppCompatActivity() {

    var count = 0
    lateinit var tvCount: TextView
    lateinit var btnCount: FloatingActionButton

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tvCount = findViewById(R.id.text_count)
        btnCount = findViewById(R.id.count_btn)
    }

    override fun onStart() {
        super.onStart()
        btnCount.setOnClickListener {
            ++count
            updateTvCount()
        }
        updateTvCount()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putInt("last_count",count)
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        count = savedInstanceState.getInt("last_count")
        updateTvCount()
    }

    fun updateTvCount() {
        tvCount.text = count.toString()
    }
}